backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -50,4 +54,19 @@
   public Staff createStaff(Staff newStaff) {
     return staffRepository.save(newStaff);
   }
+
+  /**
+   * Checks to see if the current user is a manager.
+   *
+   * @param authString the current threads authentication token.
+   * @return true if user has manager permissions.
+   */
+  public boolean hasManagerPermissions(String authString) {
+    final Role role = AuthenticationUtils
+        .getUserRoleFromToken(authString)
+        .orElseThrow(() ->
+            new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Failed to authorize.")
+        );
+    return role == Role.MANAGER;
+  }
```

Comment:
I would move this to AuthenticationUtils.java

Reply:
I was thinking this, but wasn't sure if it belonged there or not. I guess since it still takes an `authString` as a parameter it should all be handled in `AuthUtils`.

Reply:
Should the `StaffService` still contain a `hasManagerPermissions()` method? Because of the way spring security handles an auth session, we can access the access token for the current execution thread at any time by using:

```SecurityContextHolder.getContext().getAuthentication()```

This could be done in `AuthenticationUtils`.

Reply:
I don't feel very strongly about this. 
The way the classes work in my head is as follows:
AuthenticationUtils - all authentication/permission related methods
StaffController - all http REST-related methods
StaffService - helper methods for StaffController, reading from the database

This particular method is a mixture of AuthenticationUtils (`return role == Role.MANAGER`) and StaffController (`ResponseStatusException`), so I'd split it in 2 and put them in their respective areas.

I'm not too bothered about the way it's written now though.